tests/integration/test_refreshable_mv_skip_old_temp_table_ddls/test.py

```diff
@@ -0,0 +1,160 @@
+import datetime
+import logging
+import random
+import string
+import time
+from random import randint
+
+import pytest
+
+from helpers.cluster import ClickHouseCluster, QueryRuntimeException
+from helpers.network import PartitionManager
+from helpers.test_tools import assert_eq_with_retry, assert_logs_contain
+
+cluster = ClickHouseCluster(__file__)
+
+node1 = cluster.add_instance(
+    "node1",
+    main_configs=["configs/config.xml"],
+    user_configs=["configs/users.xml"],
+    with_zookeeper=True,
+    with_minio=True,
+    keeper_required_feature_flags=["multi_read", "create_if_not_exists"],
+    macros={"shard": "shard1", "replica": "1"},
+    stay_alive=True,
+)
+node2 = cluster.add_instance(
+    "node2",
+    main_configs=["configs/config.xml"],
+    user_configs=["configs/users.xml"],
+    with_zookeeper=True,
+    keeper_required_feature_flags=["multi_read", "create_if_not_exists"],
+    macros={"shard": "shard1", "replica": "2"},
+    stay_alive=True,
+)
+
+
+@pytest.fixture(scope="module")
+def started_cluster():
+    try:
+        cluster.start()
+        yield cluster
+
+    finally:
+        cluster.shutdown()
+
+
+def count_skip_ddls(node, db_name, since_ts):
+    node.query("SYSTEM FLUSH LOGS")
+    result = node.query(
+        f"SELECT count() FROM system.text_log WHERE logger_name='DDLWorker({db_name})' AND position(message, 'Skip DDL query') > 0 AND event_time_microseconds > '{since_ts}'"
+    ).strip()
+    return int(result)
+
+def get_random_string(string_length=8):
+    alphabet = string.ascii_letters + string.digits
+    return "".join((random.choice(alphabet) for _ in range(string_length)))
+
+def get_last_ddl_worker_log_ts(node, db_name):
+    node.query("SYSTEM FLUSH LOGS")
+    result = node.query(
+        f"SELECT max(event_time_microseconds) FROM system.text_log WHERE logger_name='DDLWorker({db_name})' "
+    ).strip()
+
+    if len(result) == 0:
+        return datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+
+    return result
+
+
+@pytest.mark.parametrize("append", [True, False])
+@pytest.mark.parametrize("with_inner_table", [True, False])
+@pytest.mark.parametrize(
+    "allow_skipping_old_temporary_tables_ddls_of_refreshable_materialized_views",
+    [True, False],
+)
+def test_refreshable_mv_skip_old_temp_tables_ddls(
+    started_cluster,
+    append: bool,
+    with_inner_table: bool,
+    allow_skipping_old_temporary_tables_ddls_of_refreshable_materialized_views: bool,
+):
+    db_name = "test_" + get_random_string()
+    node1.query(f"DROP DATABASE IF EXISTS {db_name} SYNC")
+    node2.query(f"DROP DATABASE IF EXISTS {db_name}  SYNC")
+
+    allow_skipping_old_temporary_tables_ddls_of_refreshable_materialized_views_settings_value = (
+        1
+        if allow_skipping_old_temporary_tables_ddls_of_refreshable_materialized_views
+        else 0
```

Comment:
I like explanatory names, but this is a bit too long. Maybe let's just use `allow_skipping`? In the context of this test it should be enough.

Furthermore, the conversion can be done easily with `int(allow_skipping_old_temporary_tables_ddls_of_refreshable_materialized_views)` or just use 0 and 1 for the parameters. Yes, it is not the best thing, but `0` converts to `False`, `1` converts to `True` and you don't have to have to different variables for different representations.